design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+- Report success and return early
+
+#### Feature Gating
+
+Similar to all other new features being added to Crossplane, we will ship this new policy as an alpha feature that will
+be off by default and will be controlled by `--enable-alpha-management-policies` flag.
+
+This will not prevent the field from appearing in the schema of the managed resources. However, we will ignore the
+`spec.managementPolicy` when the feature is not enabled.
+
+#### Deprecation of `deletionPolicy`
+
+With the new `managementPolicy` covering the existing `deletionPolicy`, we will deprecate the latter in favor of the
+former.
+
+Until we drop the `deletionPolicy` from the schema altogether, we need to be careful with the conflicting combinations
+shown in the below table which only exists with deletion:
+
+| Deletion Policy | Management Policy | Should Observe? | Should Create? | Should Update? | Should Delete? |
+| --- | --- | --- | --- | --- | --- |
+| Delete | Full | Yes | Yes | Yes | Yes |
+| Orphan | ObserveCreateUpdate | Yes | Yes | Yes | No |
+| Delete | ObserveOnly | Yes | No | No | Conflict (No) |
+| Orphan | Full | Yes | Yes | Yes | Conflict (No) |
+| Delete | ObserveCreateUpdate | Yes | Yes | Yes | Conflict (No) |
+| Orphan | ObserveOnly | Yes | No | No | No |
+
+For conflicting cases, we will decide based on the non-default configuration and not delete the external resource.
+This way, we will also err on the side of caution by leaving the actual resource. However, this comes at the cost of
+some possible confusion for users focusing on only one of the policies.
+
+Another solution could be simply throwing an error and preventing reconciliation during conflict. This would be more
+explicit but would require some manual actions and degraded UX for the usage of the feature, for example:
+
+- Creating an ObserveOnly resource will require both setting `managementPolicy` to `ObserveOnly` and `deletionPolicy`
+to `Orphan` .
+- If there are existing resources with `deletionPolicy: Orphan` when the feature is enabled, they will start failing to
+reconcile until their `managementPolicy`’s updated to `ObserveCreateUpdate`.
+
+#### Schema Changes
+
+The proposed approach here involves utilizing the same CR, hence schema, for both managing and observing resources.
+The caveat here is that some fields are required for creating the resources but not for observing. This means that it
+won’t be possible to create an observe-only resource without providing a value for any required fields, because
+the Kubernetes API checks for the presence of these fields before allowing the CR to be created.
+
+> Please note this is already the case with [importing existing managed resources] today.
+
+We will fix this by leveraging the [Common Expression Language (CEL)], which was graduated to beta (i.e. enabled by default)
+as of Kubernetes 1.25. See the following diff for the required changes we need for making [CIDRBlock] parameter of
```

Comment:
Would this make v1.25 the minimum supported version of Kubernetes for folks who wanted to use Crossplane? 

Typically we like to support all currently supported versions of Kubernetes (the three most recent minor versions). Would that be possible - I'm wondering if we could build a webhook that handled the CEL logic for older versions.

Reply:
FWIW, XP will still be usable prior v1.25 at the cost of some UX degradation, as I detailed here: 

https://github.com/crossplane/crossplane/pull/3531#discussion_r1049608625